tests/phpunit/testcases/core/themeCompatibility.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * @group core
+ * @group core-theme-compatibility
+ */
+class BP_Tests_Theme_Compatibility_Functions extends BP_UnitTestCase {
+
+	/**
+	 * @group bp_current_theme_supports
+	 */
+	public function test_bp_current_theme_doesnot_support() {
+		$this->assertFalse( bp_current_theme_supports() );
+	}
+
+	/**
+	 * @group bp_current_theme_supports
+	 */
+	public function test_bp_current_theme_does_support_buddypress() {
+		add_theme_support( 'buddypress' );
+		$this->assertTrue( bp_current_theme_supports() );
+		remove_theme_support( 'buddypress' );
+	}
+
+	/**
+	 * @group bp_current_theme_supports
+	 */
+	public function test_bp_current_theme_doesnot_support_buddypress_feature() {
+		add_theme_support( 'buddypress' );
+		$this->assertFalse( bp_current_theme_supports( array( 'activity' => 'feature' ) ) );
+		remove_theme_support( 'buddypress' );
+	}
+
+	/**
+	 * @group bp_current_theme_supports
+	 */
+	public function test_bp_current_theme_does_support_buddypress_feature() {
+		add_theme_support(
+			'buddypress',
+			array(
+				'activity' => array( 'feature1', 'feature2' ),
```

Comment:
Could we add a test checking if the theme supports both at the same time or multiple for different components?

Reply:
Let's keep it simple for now.

I've specifically disabled this kind of usage adding a doing it wrong. I prefer to first test we can use it to check for the `[ 'activity', 'block-editor' ]` && `[ 'messages', 'pagination' ]`. If we need more complex checks, we'll add them into 15.0.